### DISCUSSION TITLE
Database reads continually returning null or undefined

### DISCUSSION THREAD

**Discussion by mrdbourke** [supabase/supabase#5432] [Questions] [answered]
Hello,

I just started using Supabase and have been following a few of the templates.

However, I can't seem to get anything back from my tables.

It looks like the connections are fine but the information isn't coming back.

I've got a Replit showcasing the code I'm using here: https://replit.com/@mrdbourke/fooddatacentralsupabasetest 

Or in JavaScript:

```javascript
const { createClient } = require("@supabase/supabase-js")

const SUPABASE_URL = "https://nqicivptemixgpdaexsl.supabase.co"
const SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im5xaWNpdnB0ZW1peGdwZGFleHNsIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NDQ2NTY3MjksImV4cCI6MTk2MDIzMjcyOX0.HdD4QSNHs8_0b8c2viEq4ulIHaPVgZ5dypmcCF-mVMw"

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY)

async function main() {
  let { data, error } = await supabase
    .from("food_data_central_nutrition_data_test")
    .select("fdc_id", "food")
    .order("food", {ascending: true})

  if (error) {
    console.log("logging error");
    console.log(error);
    return;
  }
  console.log("logging data");
  console.log(data);
}

main()
```

Running this logs "logging data" but just an empty array after that.

For reference, this is the guide I'm using: https://dev.to/supabase/how-to-use-supabase-in-replit-with-node-js-3jn8

**Comment by GaryAustin1** [comment, marked as answer]
You likely have RLS with no policy or one that is not met.   Empty array is returned on select if not met.

**Comment by mrdbourke** [comment]
Thank you! I managed to fix it by setting a row-level security (RLS) policy of public read-all.

Cheers

**Reply by musaad-alosaimi** [reply]
Thanks that solved it in my side,

**Comment by abd-ulbasit** [comment]
wound't be better if proper response is returned by the supabase server instead of empty array.. I was banging my head before coming here..

**Reply by GaryAustin1** [reply]
This is the way the database Postgres operates with RLS for security.  It is silent on selects because it is just eliminating rows you can't see.  You might still get back some rows or none depending on what you are allowed to see.  It is the same as if you apply a filter and no rows match that.  The security aspect is to keep someone from "probing" the data looking for errors.
